packages/server-wallet/src/wallet/__test__/signing-address.test.ts

```diff
@@ -0,0 +1,34 @@
+import {ethers} from 'ethers';
+
+import {truncate} from '../../db-admin/db-admin-connection';
+import knex from '../../db/connection';
+import {Store} from '../store';
+import {seedAlicesSigningWallet} from '../../db/seeds/1_signing_wallet_seeds';
+import {SigningWallet} from '../../models/signing-wallet';
+
+import {alice} from './fixtures/participants';
+
+beforeEach(async () => {
+  await truncate(knex);
+});
+
+describe('signingAddress', () => {
```

Comment:
Can you add a test, which might flicker, but almost certainly won't, that makes sure that it can't add two signing addresses?

Perhaps even better: what about adding a DB constraint that ensures there is at most one signing address? One easy way is to add a column whose value must be `0`, and add a uniqueness constraint on that column. With a tactic like that, we don't need to use transactions -- you can just try-catch inserting a random wallet, and on that constraint violation, fetch the wallet that you now know exists.